src/modules/auth/hooks/useRestrictWindowActions.js

```diff
@@ -0,0 +1,14 @@
+import { useCallback, useEffect } from 'react';
+
+export const useRestrictWindowActions = ({ actions, window }) => {
+  const restrictAction = useCallback((event) => {
+    event.preventDefault();
+  }, []);
+
+  useEffect(() => {
+    actions?.forEach((action) => action && window.addEventListener(action, restrictAction));
+    return () => {
+      actions?.forEach((action) => action && window?.removeEventListener(action, restrictAction));
```

Comment:
I feel like `window` would still exist when the hook is unmounted, hence no need for the optional chaining, but other than that the hook works as expected